subprojects/core/src/main/java/org/gradle/api/internal/file/copy/NormalizingCopyActionDecorator.java

```diff
@@ -142,112 +143,138 @@ public long getLastModified() {
+            throw new UnsupportedOperationException("It's a synthetic FileCopyDetails just to create parent directories");
         }
 
         @Override
         public boolean isDefaultDuplicatesStrategy() {
-            throw new UnsupportedOperationException();
+            throw new UnsupportedOperationException("It's a synthetic FileCopyDetails just to create parent directories");
         }
 
         @Override
         public String getSourceName() {
-            throw new UnsupportedOperationException();
+            throw new UnsupportedOperationException("It's a synthetic FileCopyDetails just to create parent directories");
         }
 
         @Override
         public String getSourcePath() {
-            throw new UnsupportedOperationException();
+            throw new UnsupportedOperationException("It's a synthetic FileCopyDetails just to create parent directories");
         }
 
         @Override
         public RelativePath getRelativeSourcePath() {
-            throw new UnsupportedOperationException();
+            throw new UnsupportedOperationException("It's a synthetic FileCopyDetails just to create parent directories");
         }
 
         @Override
         public ContentFilterable filter(Map<String, ?> properties, Class<? extends FilterReader> filterType) {
-            throw new UnsupportedOperationException();
+            throw new UnsupportedOperationException("It's a synthetic FileCopyDetails just to create parent directories");
         }
 
         @Override
         public ContentFilterable filter(Class<? extends FilterReader> filterType) {
-            throw new UnsupportedOperationException();
+            throw new UnsupportedOperationException("It's a synthetic FileCopyDetails just to create parent directories");
         }
 
         @Override
         public ContentFilterable filter(Closure closure) {
-            throw new UnsupportedOperationException();
+            throw new UnsupportedOperationException("It's a synthetic FileCopyDetails just to create parent directories");
         }
 
         @Override
         public ContentFilterable filter(Transformer<String, String> transformer) {
-            throw new UnsupportedOperationException();
+            throw new UnsupportedOperationException("It's a synthetic FileCopyDetails just to create parent directories");
         }
 
         @Override
         public ContentFilterable expand(Map<String, ?> properties) {
-            throw new UnsupportedOperationException();
+            throw new UnsupportedOperationException("It's a synthetic FileCopyDetails just to create parent directories");
         }
 
         @Override
         public ContentFilterable expand(Map<String, ?> properties, Action<? super ExpandDetails> action) {
-            throw new UnsupportedOperationException();
+            throw new UnsupportedOperationException("It's a synthetic FileCopyDetails just to create parent directories");
```

Comment:
:thinking: I wonder if it is worth it to extract a method to construct the exception, so we don't have to replicate the message dozen times.